addons/payment_test/models/payment_token.py

```diff
@@ -0,0 +1,15 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details
+
+from odoo import fields, models
+
+class PaymentToken(models.Model):
+    _inherit = 'payment.token'
+
+    # State must correspond to the existing state of payment transaction. 'Draft' and 'Authorized'
+    # are ignored since payment by token are captured immediately and it is not probable to have a
+    # token that creates 'Draft' transactions.
+    state = fields.Selection(
+        selection=[('pending', "Pending"), ('done', "Confirmed"), ('cancel', "Canceled"),
+                   ('error', "Error")],
+        string="Status", default='done', required=True,
+        help="This status will be applied to all transactions made with this token",)
```

Comment:
I don't understand the use of this field, and even less why you are adding a stored required field on all tokens :thinking:.

Reply:
Okok, some points of improvement though:
1) rename the field, must be clear that it's a field from the payment_test module, for the Test acquirer.
--> test_state or anything else, but be more specific
--> Please make it clear in the string and the help that it's a test status, for payments done with the test acquirer.  We do not want studio users to show this field and use it, wrongly thinking it means something else.
2) Remove the required & the default, we don't want useless values in non test tokens (nor migration script/auto population of the column on upgrade).  Only make sure it's correctly managed by the 'payment_test' flows, with a fallback on 'done' if needed.